src/ParallelAlgorithms/EventsAndTriggers/Completion.hpp

```diff
@@ -36,9 +42,16 @@ class Completion : public Event {
                   const ArrayIndex& array_index,
                   const Component* const /*meta*/,
                   const ObservationValue& /*observation_value*/) const {
-    auto al_gore = Parallel::local(
-        Parallel::get_parallel_component<Component>(cache)[array_index]);
-    al_gore->set_terminate(true);
+    if constexpr (Parallel::is_nodegroup_v<Component>) {
+      Parallel::local_synchronous_action<
+          Parallel::Actions::SetTerminateOnElement>(
```

Comment:
This action hasn't been written yet, correct?

Reply:
Written but will be merged in a separate PR :)